myceli/src/config.rs

```diff
@@ -1,52 +1,22 @@
-use serde_derive::Deserialize;
-use tracing::info;
+use anyhow::Result;
+use figment::{
+    providers::{Format, Serialized, Toml},
+    Figment,
+};
+use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Deserialize)]
-struct RawMyceliConfig {
-    pub listen_address: Option<String>,
-    pub retry_timeout_duration: Option<u64>,
-    pub storage_path: Option<String>,
-    pub mtu: Option<u16>,
-}
-
-pub struct MyceliConfig {
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Config {
     pub listen_address: String,
     pub retry_timeout_duration: u64,
     pub storage_path: String,
     pub mtu: u16,
+    pub window_size: u8,
 }
 
-impl MyceliConfig {
-    pub fn parse(path: Option<String>) -> Self {
-        let default_config = MyceliConfig::default();
-        if let Some(path) = path {
-            info!("Myceli running config values from {path}");
-            let parsed_cfg = std::fs::read_to_string(path).expect("Failed to read config file");
-            let parsed_cfg: RawMyceliConfig =
-                toml::from_str(&parsed_cfg).expect("Failed to parse config file");
-
-            return Self {
-                listen_address: parsed_cfg
-                    .listen_address
-                    .unwrap_or(default_config.listen_address),
-                retry_timeout_duration: parsed_cfg
-                    .retry_timeout_duration
-                    .unwrap_or(default_config.retry_timeout_duration),
-                storage_path: parsed_cfg
-                    .storage_path
-                    .unwrap_or(default_config.storage_path),
-                mtu: parsed_cfg.mtu.unwrap_or(default_config.mtu),
-            };
-        } else {
-            info!("Myceli running default config values");
-        }
-        default_config
-    }
-}
-
-impl std::default::Default for MyceliConfig {
+impl Default for Config {
     fn default() -> Self {
-        Self {
+        Config {
```

Comment:
Why change from Self to Config?

Reply:
Hmm I don't think there was a strong reason to do this, other than I had copied a new config pattern from `hyphae` and in that pattern I used `Config` instead of `Self`.